SatisfactorySaveEditor/ViewModel/MainViewModel.cs

```diff
@@ -249,23 +270,28 @@ private void Save(bool saveAs)
                     HasUnsavedChanges = false;
                     RaisePropertyChanged(() => FileName);
                     AddRecentFileEntry(dialog.FileName);
+                    this.IsBusy = false;
                 }
             }
             else
             {
-                AutoBackupIfEnabled();
-                
+                this.IsBusy = true;
+                await Task.Run(() => AutoBackupIfEnabledAsync());
+                this.IsBusy = false;
+
                 var newObjects = rootItem.DescendantSelf;
                 saveGame.Entries = saveGame.Entries.Intersect(newObjects).ToList();
                 saveGame.Entries.AddRange(newObjects.Except(saveGame.Entries));
 
                 rootItem.ApplyChanges();
-                saveGame.Save();
+                this.IsBusy = true;
+                await Task.Run(() => saveGame.Save());
+                this.IsBusy = false;
                 HasUnsavedChanges = false;
             }
         }
 
-        private void AutoBackupIfEnabled()
+        private void AutoBackupIfEnabledAsync()
```

Comment:
You put Async in the name but it's not actually async?

Reply:
Resolved in next commits